packages/block-editor/src/utils/get-paste-event-data.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { find, isNil } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createBlobURL } from '@wordpress/blob';
+
+export function getPasteEventData( { clipboardData } ) {
+	let { items, files } = clipboardData;
+
+	// In Edge these properties can be null instead of undefined, so a more
+	// rigorous test is required over using default values.
+	items = isNil( items ) ? [] : items;
+	files = isNil( files ) ? [] : files;
+
+	let plainText = '';
+	let html = '';
+
+	// IE11 only supports `Text` as an argument for `getData` and will
+	// otherwise throw an invalid argument error, so we try the standard
+	// arguments first, then fallback to `Text` if they fail.
+	try {
+		plainText = clipboardData.getData( 'text/plain' );
+		html = clipboardData.getData( 'text/html' );
+	} catch ( error1 ) {
+		try {
+			html = clipboardData.getData( 'Text' );
+		} catch ( error2 ) {
+			// Some browsers like UC Browser paste plain text by default and
+			// don't support clipboardData at all, so allow default
+			// behaviour.
+			return;
+		}
+	}
```

Comment:
This is a super thorough block in terms of browser support :), what did you base it on?

Reply:
Some of this is actually a copy from rich text. :) We'll need to find a way to allow this function to be reused.